### DISCUSSION TITLE
Hide the addRowsComponent

### DISCUSSION THREAD

**Discussion by loztech** [nick-keller/react-datasheet-grid#174] [Q&A] [answered]
How would I hide/remove the addRowsComponent?

It seems that this should have worked:
      addRowsComponent={undefined}

Thanks.

**Comment by nick-keller** [comment]
Hello @loztech,
Passing `undefined` is the same as not specifying a value, and not specifying a value results in the default value being used.

You could pass `lockRows={false}` if that is what you are trying to do. ([doc](https://react-datasheet-grid.netlify.app/docs/api-reference/props#lockrows))

Otherwise what you can do is passe an empty component:

```tsx
const Emptycomponent = () => {
  return null
}

const App = () => {
  return (
    <DataSheetGrid
      addRowsComponent={Emptycomponent}
    />
  ) 
}
```

I'll admit this is a bit convoluted. I will make sur you can pass `null` or `false` to disable it more easily.

**Reply by loztech** [reply]
Ah yes, `lockRows={true}` does hide the addRowsComponent at the bottom of the datasheet, which is what I was trying to accomplish.  And this makes sense, as it supports "prevent[ing] the user from **adding** or removing rows".   

The `{Emptycomponent}` approach would be necessary if I wanted ContextMenus to still work, but hide the addRowsComponent from the datasheet.

Many thanks!

**Comment by nick-keller** [comment, marked as answer]
Either use `lockRows={false}` to prevent the user from ever adding / removing rows.
or use `addRowsComponent={false}` to still let the user add / delete rows using right click, but hiding the add row widget at the bottom
